tests/net/route_mcast/src/main.c

```diff
@@ -0,0 +1,665 @@
+}
+
+static void net_route_mcast_iface_init2(struct net_if *iface)
+{
+	iface_2 = iface;
+	net_route_mcast_add_addresses(iface, &iface_2_addr, &ll_addr_2);
+}
+
+static void net_route_mcast_iface_init3(struct net_if *iface)
+{
+	iface_3 = iface;
+	net_route_mcast_add_addresses(iface, &iface_3_addr, &ll_addr_3);
+}
+
+static bool check_packet_addresses(struct net_pkt *pkt)
+{
+	struct net_ipv6_hdr *ipv6_hdr = NET_IPV6_HDR(pkt);
+
+	if ((memcmp(&active_scenario.src,
+			&ipv6_hdr->src,
+			sizeof(struct in6_addr)) != 0) ||
+			(memcmp(&active_scenario.mcast,
+					&ipv6_hdr->dst,
+					sizeof(struct in6_addr)) != 0)) {
+		return false;
+	}
+
+	return true;
+}
+static int iface_send(struct device *dev, struct net_pkt *pkt)
+{
+	if (!active_scenario.is_active) {
+		return 0;
+	}
+	if (!check_packet_addresses(pkt)) {
+		return 0;
+	}
+
+	forwarding_counter++;
+
+	if (net_pkt_iface(pkt) == iface_1) {
+		iface_1_forwarded = true;
+	} else if (net_pkt_iface(pkt) == iface_2) {
+		iface_2_forwarded = true;
+	} else if (net_pkt_iface(pkt) == iface_3) {
+		iface_3_forwarded = true;
+	}
+
+	return 0;
+}
+
+struct net_route_mcast_iface_cfg net_route_data_if1;
+struct net_route_mcast_iface_cfg net_route_data_if2;
+struct net_route_mcast_iface_cfg net_route_data_if3;
+
+static struct dummy_api net_route_mcast_if_api_1 = {
+	.iface_api.init = net_route_mcast_iface_init1,
+	.send = iface_send,
+};
+
+static struct dummy_api net_route_mcast_if_api_2 = {
+	.iface_api.init = net_route_mcast_iface_init2,
+	.send = iface_send,
+};
+
+static struct dummy_api net_route_mcast_if_api_3 = {
+	.iface_api.init = net_route_mcast_iface_init3,
+	.send = iface_send,
+};
+
+#define _ETH_L2_LAYER DUMMY_L2
+#define _ETH_L2_CTX_TYPE NET_L2_GET_CTX_TYPE(DUMMY_L2)
+
+NET_DEVICE_INIT_INSTANCE(mcast_iface_1, "mcast_iface_1", iface_1,
+		net_route_mcast_dev_init, device_pm_control_nop,
+			 &net_route_data_if1, NULL,
+			 CONFIG_KERNEL_INIT_PRIORITY_DEFAULT,
+			 &net_route_mcast_if_api_1, _ETH_L2_LAYER,
+			 _ETH_L2_CTX_TYPE, 127);
+
+NET_DEVICE_INIT_INSTANCE(mcast_iface_2, "mcast_iface_2", iface_2,
+		net_route_mcast_dev_init, device_pm_control_nop,
+			 &net_route_data_if2, NULL,
+			 CONFIG_KERNEL_INIT_PRIORITY_DEFAULT,
+			 &net_route_mcast_if_api_2, _ETH_L2_LAYER,
+			 _ETH_L2_CTX_TYPE, 127);
+
+NET_DEVICE_INIT_INSTANCE(mcast_iface_3, "mcast_iface_3", iface_3,
+		net_route_mcast_dev_init, device_pm_control_nop,
+			 &net_route_data_if3, NULL,
+			 CONFIG_KERNEL_INIT_PRIORITY_DEFAULT,
+			 &net_route_mcast_if_api_3, _ETH_L2_LAYER,
+			 _ETH_L2_CTX_TYPE, 127);
+
+static struct net_pkt *setup_ipv6_udp(struct net_if *iface,
+				      struct in6_addr *src_addr,
+				      struct in6_addr *remote_addr,
+				      uint16_t src_port, uint16_t remote_port)
+{
+	static const char payload[] = "foobar";
+	struct net_pkt *pkt;
+	int res;
+
+	pkt = net_pkt_alloc_with_buffer(iface, strlen(payload), AF_INET6,
+					IPPROTO_UDP, K_FOREVER);
+
```

Comment:
this empty line can be removed, we usually do not have have empty line between setting a var and then checking its value